src/Workspaces/Remote/Core/RemoteEndPoint.cs

```diff
@@ -191,16 +191,20 @@ public async Task<T> InvokeAsync<T>(string targetName, IReadOnlyList<object?> ar
                 // if invoke throws an exception, make sure we raise cancellation.
                 RaiseCancellationIfInvokeFailed(task, linkedCancellationSource, cancellationToken);
 
-                // wait for asset source to respond
-                await pipe.WaitForConnectionAsync(linkedCancellationSource.Token).ConfigureAwait(false);
+                var task2 = Task.Run(async () =>
+                {
+                    // wait for asset source to respond
+                    await pipe.WaitForConnectionAsync(linkedCancellationSource.Token).ConfigureAwait(false);
 
-                // run user task with direct stream
-                var result = await dataReader(stream, linkedCancellationSource.Token).ConfigureAwait(false);
+                    // run user task with direct stream
+                    return await dataReader(stream, linkedCancellationSource.Token).ConfigureAwait(false);
+                });
```

Comment:
CancellationToken not passed to Task.Run.  If not intentional, fix.  If intentional please pass CancellationToken.NOne and doc why this is correct logic here.  

--

In other words i can't tell if this is purposeful, or an accidental omission.  That is hard to deal with later when someone comes back to this.

Reply:
➡️ Added cancellation token